### PR TITLE
cpu-freq: Try powersave if ondemand is not available

### LINKED ISSUE #9611: cpupower needs modprobe
I noticed the following errors in my journal:

```
Sep 02 17:58:46 hera systemd[1]: Starting CPU Frequency Governor Setup...
Sep 02 17:58:46 hera cpupower[479]: sh: modprobe: command not found
Sep 02 17:58:46 hera cpupower[479]: Setting cpu: 0
Sep 02 17:58:46 hera cpupower[479]: Error setting new values. Common errors:
Sep 02 17:58:46 hera cpupower[479]: - Do you have proper administration rights? (super-user?)
Sep 02 17:58:46 hera cpupower[479]: - Is the governor you requested available and modprobed?
Sep 02 17:58:46 hera cpupower[479]: - Trying to set an invalid policy?
Sep 02 17:58:46 hera cpupower[479]: - Trying to set a specific frequency, but userspace governor is not available,
Sep 02 17:58:46 hera cpupower[479]: for example because of hardware which cannot be set to a specific frequency
Sep 02 17:58:46 hera cpupower[479]: or because the userspace governor isn't loaded?
```

This is probably fixed by adding the following to `systemd.services.cpufreq` in `nixos/modules/tasks/cpu-freq.nix`:

``` nix
{ path = [ config.system.sbin.modprobe ]; }
```

(Note I'm on a recent checkout of `nixos-15.09-small`)

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

This should fix default governor picking on Sandy Bridge+ chips and close https://github.com/NixOS/nixpkgs/issues/9611
Not all people monitor their CPUs at least eventually. And we don't want them to shorten lives of their machines.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by gnidorah** [comment]
Probably figuring correct governor on install stage and setting it at hardware-configuration.nix should be a better approach (if that's possible).

**Comment by joachifm** [comment]
See e.g., `/sys/devices/system/cpu/cpufreq/*/scaling_available_governors`.

**Comment by gnidorah** [comment]
@joachifm Okay, let's try this 2c8253c325834e16251936e994c100e893d79253

File: nixos/modules/config/power-management.nix [hunk 1]
```diff
@@ -69,9 +69,6 @@ in
 
   config = mkIf cfg.enable {
 
-    # FIXME: Implement powersave governor for sandy bridge or later Intel CPUs
-    powerManagement.cpuFreqGovernor = mkDefault "ondemand";
-
```
Comment (grahamc) [inline nixos/modules/config/power-management.nix]: Probably should keep this, since almost nobody reruns the config generator.

Comment (gnidorah) [inline nixos/modules/config/power-management.nix]: @grahamc Fixed, thanks!

File: nixos/modules/tasks/cpu-freq.nix [hunk 2]
```diff
@@ -43,7 +43,12 @@ in
       serviceConfig = {
         Type = "oneshot";
         RemainAfterExit = "yes";
-        ExecStart = "${cpupower}/bin/cpupower frequency-set -g ${cfg.cpuFreqGovernor}";
+        ExecStart = ''
+          ${pkgs.bash}/bin/sh -c \
+          "for i in ${toString cfg.cpuFreqGovernor}; do \
+            ${cpupower}/bin/cpupower frequency-set -g $i && break; \
+          done"
```
Comment (Mic92) [inline nixos/modules/tasks/cpu-freq.nix]: you can use `script` instead of `serviceConfig.ExecStart` to achieve the same.

**Pull request merged**